### PR TITLE
Improve cifar10_cnn_capsule.py

### PR DESCRIPTION
**Pull request by Sklan**
### Summary
Improves the cifar10_cnn_capsule.py example.
Fixes spelling and formatting.
Removes unnecessary function imports and unneeded function declaration.
It also solves the below error by adding the steps_per_epoch argument in fit_generator.
```python
ValueError: `steps_per_epoch=None` is only valid for a generator based on the `keras.utils.Sequence` class. Please specify `steps_per_epoch` or use the `keras.utils.Sequence` class.
```
### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: examples/cifar10_cnn_capsule.py [hunk 1]
```diff
@@ -231,6 +216,7 @@ def compute_output_shape(self, input_shape):
     # Fit the model on the batches generated by datagen.flow().
     model.fit_generator(
         datagen.flow(x_train, y_train, batch_size=batch_size),
+        steps_per_epoch= len(x_train) / batch_size,
```
Comment (gabrieldemarmiesse) [inline examples/cifar10_cnn_capsule.py]: The bug you encountered is a bug appearing only in keras 2.2.4. The version from master can run this script without any issues.

File: examples/cifar10_cnn_capsule.py [hunk 2]
```diff
@@ -1,59 +1,51 @@
...
 from __future__ import print_function
-from keras import backend as K
-from keras.layers import Layer
 from keras import activations
+from keras import backend as K
 from keras import utils
 from keras.datasets import cifar10
-from keras.models import Model
-from keras.layers import *
+from keras.layers import AveragePooling2D, Conv2D, Lambda, Layer, Reshape
```
Comment (fchollet) [inline examples/cifar10_cnn_capsule.py]: Prefer importing `layers` then using e.g. `layers.Conv2D`

File: examples/cifar10_cnn_capsule.py [hunk 3]
```diff
@@ -1,59 +1,51 @@
...
     return K.sum(y_true * K.square(K.relu(1 - margin - y_pred)) + lamb * (
         1 - y_true) * K.square(K.relu(y_pred - margin)), axis=-1)
 
 
 class Capsule(Layer):
-    """A Capsule Implement with Pure Keras
-    There are two vesions of Capsule.
-    One is like dense layer (for the fixed-shape input),
-    and the other is like timedistributed dense (for various length input).
+    """A Capsule Network implementation in Keras
```
Comment (fchollet) [inline examples/cifar10_cnn_capsule.py]: Please standardize the docstrings formatting to be the same as other docstrings in the codebase.

File: examples/cifar10_cnn_capsule.py (hunk 1 above)
Comment (Sklan) [reply examples/cifar10_cnn_capsule.py]: It seems that on updating to master the script fails at the batch_dot operation.
```python 'y.shape[%d] (%d != %d).' % (axes[0], axes[1], d1, d2))
ValueError: Can not do batch_dot on inputs with shapes (None, 10, 10, 16) and (None, 10, None, 16) with axes=[2, 3]. x.shape[2] != y.shape[3] (10 != 16).
```

Comment (gabrieldemarmiesse) [reply examples/cifar10_cnn_capsule.py]: @farizrahman4u could this be related to tour recent modifications of batch dot?

**Pull request merged**